File: AndroidToolbarWorking/app/src/main/java/com/weatherlogger/androidtoolbarworking/Deneme1.java
package com.weatherlogger.androidtoolbarworking;

import android.graphics.Color;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import java.util.Objects;

public class Deneme1 extends AppCompatActivity {

    private MenuItem menuItem;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.deneme1);

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolBar);

        // For Toolbar Text..
        toolbar.setTitle("Fatih Koçak");
        toolbar.setTitleTextColor(Color.parseColor("#FFFFFF"));
        setSupportActionBar(toolbar);
        Objects.requireNonNull(getSupportActionBar()).setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setDisplayShowHomeEnabled(true);
        /*getSupportActionBar().setDisplayShowTitleEnabled(false);
        toolbar.setNavigationIcon(R.drawable.ic_chevron_left);*/

        toolbar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(getApplicationContext(), "Toolbar' a tıklandı", Toast.LENGTH_SHORT).show();
            }
        });
    }

    @Override
    public boolean onSupportNavigateUp() {
        onBackPressed();
        return true;
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu, menu);
        menuItem = menu.findItem(R.id.action_search);
        return true;
    }

    @Override
    public boolean onPrepareOptionsMenu(Menu menu) {
        super.onPrepareOptionsMenu(menu);
        menuItem.setIcon(R.drawable.ic_chevron_left);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                onBackPressed();
                break;
            case R.id.action_search:
                onBackPressed();
                Toast.makeText(getApplicationContext(), "Action Search", Toast.LENGTH_SHORT).show();
                break;
            case R.id.action_like:
                Toast.makeText(getApplicationContext(), "Action Like", Toast.LENGTH_SHORT).show();
                break;
        }
        return true;
    }
}

File: AndroidWorkingToolBarKotlin/app/src/main/java/com/weatherlogger/androidworkingtoolbarkotlin/Deneme1.kt
package com.weatherlogger.androidworkingtoolbarkotlin

import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.Toast
import androidx.appcompat.app.ActionBar
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import kotlinx.android.synthetic.main.deneme1.*
import java.util.*

class Deneme1 : AppCompatActivity() {

    private var isPlaying: Boolean = false
    //private lateinit var menuItem: MenuItem

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.deneme1)

        setSupportActionBar(toolBar as Toolbar?)
        Objects.requireNonNull<ActionBar>(supportActionBar).setDisplayHomeAsUpEnabled(true)
        supportActionBar!!.setDisplayShowHomeEnabled(true)
        supportActionBar!!.setDisplayShowTitleEnabled(false)
        (toolBar as Toolbar?)?.setNavigationIcon(R.drawable.ic_chevron_left)

        toolBar.setOnClickListener {
            Toast.makeText(applicationContext, "Toolbar' a tıklandı", Toast.LENGTH_SHORT).show()
        }
    }

    private fun changeToolbar(){
        invalidateOptionsMenu()
    }

    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return true
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu, menu)

        if (!isPlaying)
            menu.removeItem(R.id.action_search2)
        else
            menu.removeItem(R.id.action_search)
        return true
    }

    override fun onPrepareOptionsMenu(menu: Menu): Boolean {
        super.onPrepareOptionsMenu(menu)

        if (!isPlaying)
            menu.findItem(R.id.action_search).setIcon(R.drawable.ic_chevron_left)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> onBackPressed()
            R.id.action_search -> {
                //onBackPressed()
                isPlaying = true
            }
            R.id.action_search2 -> {
                //onBackPressed()
                isPlaying = false
            }
            R.id.action_like -> Toast.makeText(
                applicationContext,
                "Action Like",
                Toast.LENGTH_SHORT
            ).show()
        }

        changeToolbar()

        return true
    }
}